lib/crawler.js

```diff
@@ -377,14 +377,18 @@ class Crawler {
         throw new Error('Delete without metadata');
       }
       request.document._metadata = request.response._metadataTemplate;
+      if (request.context.history) {
+        request.document._metadata.rootRequest = request.context.history[0];
```

Comment:
so, request.context.history array keeps track of the full event processing graph? in other words, if there was an event that triggered a repo pull, which triggered an issue pull, would history be:
- [0] event (main GitHub request)
- [1] repo pull request
- [2] issue pull request

Again, just for my understanding.

Reply:
Yes, exactly. The history is only stored in queue messages. The initial purpose was to see the processing graph in failed/deadletter messages. Here's a real example from a deadletter message:
```
    "history": [
      "event_trigger@/orgs/aspnet/events/eb099530-dfb1-11e9-9dc2-43d166fadd98",
      "update_events@/orgs/aspnet/events",
      "IssueCommentEvent@/orgs/aspnet/events/10494070190",
      "issue@/repos/aspnet/AspNetCore/issues/14416",
      " Requeued: Throttled GitHub primary throttling caught us",
      "issue@/repos/aspnet/AspNetCore/issues/14416",
      " Requeued: Throttled GitHub primary throttling caught us",
      "issue@/repos/aspnet/AspNetCore/issues/14416",
      " Requeued: Throttled GitHub primary throttling caught us",
      "issue@/repos/aspnet/AspNetCore/issues/14416",
      " Requeued: Throttled GitHub primary throttling caught us",
      "issue@/repos/aspnet/AspNetCore/issues/14416",
      " Requeued: Throttled GitHub primary throttling caught us",
      "issue@/repos/aspnet/AspNetCore/issues/14416",
      " Requeued: Throttled GitHub primary throttling caught us"
    ]
```